### DISCUSSION TITLE
Weird behavior with filters in users datatable

### DISCUSSION THREAD

**Discussion by patriksh** [mindstellar/shopclass#414] [Q&A] [answered]
users_processing_row - first and second argument are the same, instead of second one being the user row?

datatable_user_class - adding any function to this filters breaks the user row argument which returns in all rows being blocked, since

```php
    private function get_row_status($user)
    {

        if ($user['b_enabled'] == 0) {
            return array(
                'class' => 'status-blocked',
                'text'  => __('Blocked')
            );
        }
    ...
```

Any ideas what might be happening? This is with Osclass 5.0.2.
All worked fine on 3.9. I'm a bit lost and any pointers could help.

**Comment by patriksh** [comment]
```
function userTableRow($row, $aRow) {
    $row['plugin_profilepic'] = print_r($aRow, 1);

    return $row;
}
osc_add_hook('users_processing_row',  'userTableRow');
```

![image](https://user-images.githubusercontent.com/53878055/135854660-91500001-2ecb-4a74-a319-804b9bec59a6.png)

**Comment by navjottomer** [comment, marked as answer]
Test this on 5.1 it'll work.

**Reply by patriksh** [reply]
> Test this on 5.1 it'll work.

Mind sending what changed in 5.1 so I can apply it to the site?

**Reply by navjottomer** [reply]
Look for plugins class for changes.

**Reply by patriksh** [reply]
Thanks, works great!

**Comment by nikk0s** [comment]
Hi!

Have the same issue with 2 plugins handling the same hook filter 'users_processing_row'

In called method : `Plugins::applyFilter($hook, $content, ...$args);`
the line :
`63:    $args[0] = $content;`
replace 3rd parameter '$aRow' (which contains initially a full user data row) by the output of the first call of the plugin hook method.
Of course this cause unexpected parameter data for the 2nd plugin call

Simply comment the line solve my problem, but may cause unexpected behavior in other place (?)
Is this line usefull ?

Thanks for your attention and the great job :)

Nicolas

**Reply by nikk0s** [reply]
I just see that this issue has been fixed in the develop branch...
Thanks !
